docs/topics/development/search.rst

```diff
@@ -0,0 +1,92 @@
+.. _search:
+
+============================
+How does search on AMO work?
+============================
+
+.. note::
+
+  This is documenting our current state of how search is implemented in addons-server.
+  We will be using this to plan future improvements so please note that we are
+  aware that the process written below is not perfect and has bugs here and there.
+
+  Please see https://github.com/orgs/mozilla/projects/17#card-10287357 for more planning.
+
+
+General structure
+=================
+
+Our search contains these data types:
```

Comment:
Technically, we only have add-ons and add-on compatibility reports documents. The rest is just children of add-on documents, so maybe make that more explicit. (Also worth using ES terminology here, even though [things are changing](https://www.elastic.co/guide/en/elasticsearch/reference/6.x/removal-of-types.html) so maybe just mention indices and not types)